docs/content/concepts/authentication/workspace.md

```diff
@@ -0,0 +1,143 @@
+
+Workspace types then reference a set of auth configs, and their configuration will apply to all their workspaces/logicalclusters. Compared to many other settings in a `WorkspaceType` that work only as a preset for _new_ workspaces, the configured auth configs will continue to affect workspaces, so when a `WorkspaceType` is changed, this will impact existing workspaces, too.
+
+For every incoming HTTPS request, kcp will then resolve the logicalcluster, determine the used workspace type, assemble the list of auth configs and create an authenticator suitable for exactly the one logicalcluster targeted by the request. This workspace authenticator is an *alternative* to kcp's regular authentication (i.e. it forms a union with it).
+
+This authentication can happen in the front-proxy or on each shard individually. However only the front-proxy has a global view across all shards and will be able to reliably resolve everything necessary. The shard-local per-workspace authentication really only works on a single
+shard and requires that all of `Workspace`, `WorkspaceType`, auth configs and `LogicalClusters` are on the same local shard. Because of this, it's recommended to use the front-proxy to handle per-workspace authentication.
+
+## OIDC
+
+It is important to understand how the per-workspace authenticators interact with the global ones. Most importantly, how audiences are handled.
+
+kcp has a `--api-audiences` flag that configures the global JWT audience claim that every single JWT needs to contain in order for it to be admitted. These global audiences are also required when using per-workspace authentication.
+
+For example, suppose kcp is started with `--api-audiences=https://kcp.example.com` and there is a `WorkspaceAuthenticationConfiguration` that defines a JWT validator using the audience `https://corp.initech.com`. For a token to be admitted into a workspace that uses this auth config, the token will have to contain *both* audiences. This is to ensure the token is actually meant to be used in kcp, regardless of which audiences are then configured per workspace.
+
+## Limitations
+
+This feature has some small limitations that users should keep in mind:
+
+* As mentioned above, the JWT validation for a workspace is not 100% independent from the global kcp authentication: tokens will need to contain kcp's global API audience (configured with `--api-audiences`) and any audience configured in the auth configs. You cannot have a token not contain kcp's global audience.
+* `WorkspaceAuthenticationConfiguration` objects must reside in the same logicalcluster as the `WorkspaceType`.
```

Comment:
Do we have example how to make sure this happens or what does it mean from the "hands on experience"?

Reply:
For the first bullet point, yes, there are e2e tests that test that your tokens need to have kcp's global audience configured to be admitted.

For the second bullet point, it's simply the fact that references from WorkspaceTypes to AuthConfigs cannot have a path. All you can configure is the name of the AuthConfig, so they logically must be in the same logicalcluster as the WorkspaceType that is referring to them.